hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/BulkInsertInternalPartitionerFactory.java

```diff
@@ -18,15 +18,26 @@
 
 package org.apache.hudi.execution.bulkinsert;
 
+import org.apache.hudi.config.HoodieWriteConfig;
 import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.index.HoodieIndex;
 import org.apache.hudi.table.BulkInsertPartitioner;
+import org.apache.hudi.table.HoodieTable;
 
 /**
  * A factory to generate built-in partitioner to repartition input records into at least
  * expected number of output spark partitions for bulk insert operation.
  */
 public abstract class BulkInsertInternalPartitionerFactory {
 
+  public static BulkInsertPartitioner get(HoodieTable table, HoodieWriteConfig config) {
```

Comment:
This will get called in executor right? If yes, then i'm a bit worried that `RDDConsistentBucketPartitioner` has `HoodieTable` and `HoodieWriteConfig` as constructor arguments. There could be significant serialization cost.

Reply:
It will be called once for constructing the partitioner in driver. But the partitioner itself possess the `table` variable, so yes there will be serialization costs.  

The serialization is done once for each executor, so the overall overhead should be OK I guess?